java/fury-core/src/main/java/org/apache/fury/serializer/ObjectSerializer.java

```diff
@@ -784,6 +784,135 @@ static boolean readBasicObjectFieldValueFailed(
     }
   }
 
+  static Object readFieldValue(
```

Comment:
This is a rather big method; I'm not sure if HotSpot can compile this. In any case, this shouldn't be in the hot path.